src/resolvers/types.ts

```diff
@@ -0,0 +1,7 @@
+import { JsonPath } from '@stoplight/types';
+
+export type ResolvedRef<Source extends string | null = null> = {
```

Comment:
what does that null = null type do here?  Just curious I haven't seen this before.

Reply:
It's a default value.

Example:
```ts
type Foo<X extends string> = Record<X, unknown>

const value: Foo = {};
```
<img width="582" alt="image" src="https://user-images.githubusercontent.com/9273484/182945133-3021a7db-1332-48e6-9615-8e90acb21ae9.png">

vs

```ts
type Foo<X extends string = string> = Record<X, unknown>

const value: Foo = {};
```
<img width="479" alt="image" src="https://user-images.githubusercontent.com/9273484/182945215-c3bed2f6-33c6-4ebb-a89e-9dfa99189b59.png">

Reply:
Ahhh.. I was reading it as extends string | (null = null) but I should have been reading it as extends (string | null) = null - my fault lol.  That makes sense